### PR TITLE
Add an API to manage the engine config.

### LINKED ISSUE #182: api: create api for getting / setting slamengine server config
## Set server port

```
PUT /server/port
```

This will restart the server at the specified port. All communication from the client must shift over to the new port.

There is no need for an API to get the server port, since in order to call the API, you would already have to know the port.

## Get all the mount configuration at a specified path

To find out which paths represent mounted data sources, a client can use the metadata API and look for mount resources. This API is to retrieve configuration for a given mount.

```
GET /mounts/fs/[path]
```

A mount is a string which represents an Connection URI.

## Create a new mount configuration at a specified path

For this method to succeed, there must not exist any existing resource (mount or otherwise) at the specified path.

```
POST /mounts/fs/[path]
```

A mount is a string which represents an Connection URI.

## Update an existing mount at a specified path

For this method to succeed, there must exist a mount at the specified path.

```
PUT /mounts/fs/[path]
```

A mount is a string which represents an Connection URI.

## Delete an existing mount at a specified path

```
DELETE /mounts/fs/[path]
```

### PR DESCRIPTION
**Pull request by sellout**
Adds two new services – one for configuring mount points, and one for
the port.

Fixes #182.

There is some … non-FP code in server.scala. I’m happy if anyone can push it toward more FP.

### CODE REVIEW

File: web/src/main/scala/slamdata/engine/api/fs.scala [hunk 1]
```diff
@@ -246,6 +245,69 @@ class FileSystemApi(fs: FSTable[Backend]) {
     }
   }
 
+  def serverService(config: Config, reloader: Config => Task[Unit]) = {
+    corsService {
+      case req @ PUT -> Root / "port" => for {
+        body <- EntityDecoder.decodeString(req)
+        r    <- body.parseInt.fold(
+          e => NotFound(e.getMessage),
```
Comment (mossprescott) [inline web/src/main/scala/slamdata/engine/api/fs.scala]: This is a 400 but not really a 404, right? Why not just `BadRequest`?

**Comment by mossprescott** [comment]
This is missing tests and documentation (i.e. at least add something to README.md). But in the interest of unblocking the front-end guys, maybe we should merge anyway.

**Comment by JeffACarr** [comment]
yep, lots of guys in PS conf, so lets unblock so Garyb/Maxim can keep moving

On Thu, May 21, 2015 at 9:21 AM, Moss Prescott <notifications@github.com>
wrote:

> This is missing tests and documentation (i.e. at least add something to
> README.md). But in the interest of unblocking the front-end guys, maybe we
> should merge anyway.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/slamdata/slamengine/pull/725#issuecomment-104317582>.
>



-- 
*Jeff A. Carr*
*Co-Founder & CEO*
*303-717-2091*
<http://slamdata.com/>

**Comment by mossprescott** [comment]
Per conversation on IRC, gonna wait a few hours before merging. @garyb, feel free to ping me here.

**Comment by sellout** [comment]
Yeah, I'll try to update at least the read me now. Sorry, I didn't want this to wait for all the confs to end.

File: web/src/main/scala/slamdata/engine/api/fs.scala (hunk 1 above)
Comment (sellout) [inline web/src/main/scala/slamdata/engine/api/fs.scala]: I think it is a 404 – /mount/fs/hdnrbdis is not found.

**Comment by garyb** [comment]
Thanks @sellout, @mossprescott.

**Pull request merged**